src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_VOLUME = 10;
+    private static double INCREASING_MULTIPLICATOR = 1.5;
+    private int size = 0;
+    private Object[] list;
+
+    public ArrayList() {
+        list = new Object[DEFAULT_VOLUME];
+    }
+
+    public ArrayList(int size) {
+        list = new Object[size];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == list.length) {
+            increaseVolume((int) Math.round(list.length * INCREASING_MULTIPLICATOR));
```

Comment:
Replace this logic "(int) Math.round(list.length * INCREASING_MULTIPLICATOR)" inside of method